### PR TITLE
LTE Performance Analysis App.

### PR DESCRIPTION
**Pull request by SrikanthPagadarai**

### CODE REVIEW

**Comment by rgetz** [comment]
There seems to be alot of duplication in this code base - do we want to fix that now? or after merge?

File: trx_examples/streaming/LTE_PA_App/@LTEAppInternals/LTEAppInternals.m [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+                       refGrid, rxGridLow, rxGridHigh, HestLow, HestHigh);
+                   if (etm.CellRefP ~= 1) && (etm.CellRefP ~= 2) && (etm.CellRefP ~= 4)
+                       obj.StopTest = true;
+                       status = sprintf...
+                           ('Test stopped. RF loopback cable likely disconnected. For the demodulated parameter field CellRefP, the value (%d) is not one of the set (1, 2, 4).',...
+                           etm.CellRefP);
+                       app.Label.Text = {status};   
+                       app.PlayStopButtonState = ~app.PlayStopButtonState;
+                       app.PlayStopButton.Icon = which('play.png');
+                       app.StepButton.Enable = 'on';
```
Comment (tfcollins) [inline trx_examples/streaming/LTE_PA_App/@LTEAppInternals/LTEAppInternals.m:189]: This should be a list and loop to set all these parameters

File: trx_examples/streaming/LTE_PA_App/@LTEAppInternals/LTEAppInternals.m [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+                       app.TxDropDown.Enable = 'on';
+                       app.TxDropDown.Value = obj.PlutoTx.RadioID;
+                       app.RxDropDown.Enable = 'on';
+                       app.RxDropDown.Value = obj.PlutoRx.RadioID;
+                       app.dBPercentDropDown.Enable = 'on';
+                       app.TMNDropDown.Enable = 'on';
+                       app.BWDropDown.Enable = 'on';
+                       app.LOEditField.Enable = 'on';
+                       
+                       app.PBCHCheckBox.Enable = 'on';
```
Comment (tfcollins) [inline trx_examples/streaming/LTE_PA_App/@LTEAppInternals/LTEAppInternals.m:292]: Setting all these properties looks like a duplicate from above. This should be a separate method

File: trx_examples/streaming/LTE_PA_App/@LTEAppInternals/LTEAppInternals.m [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+            obj.PlutoRx();
+            obj.PlutoRx.release();
+            app.PlayStopButton.Icon = which('play.png');
+            drawnow; 
+        end          
+        
+        function killtest = stopTest(obj, app)
+            killtest = obj.StopTest;
+            if (killtest)
+                app.PlayStopButton.Enable = 'on';
```
Comment (tfcollins) [inline trx_examples/streaming/LTE_PA_App/@LTEAppInternals/LTEAppInternals.m:331]: All the 'on' sets should be in a loop

File: trx_examples/streaming/LTE_PA_App/@LTEAppInternals/PlutoRadio.m [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    if (app.NumRadios > 1)
+        if strcmp(app.StepOrPlayButton, 'step') || (frame_ind == 1)
+            obj.PlutoRx = FrequencyCorrectionRadios(obj.PlutoTx, obj.PlutoRx);
+            obj.FrequencyCorrection = obj.PlutoRx.FrequencyCorrection;
+            obj.PlutoRx.SamplesPerFrame = 2^23;
+        else
+            obj.PlutoRx.FrequencyCorrection = obj.FrequencyCorrection;
+
+            switch (app.BWDropDown.Value)
+                case '3 MHz'
```
Comment (tfcollins) [inline trx_examples/streaming/LTE_PA_App/@LTEAppInternals/PlutoRadio.m:65]: Common cases in switch statements should use bracket notation {'3 MHz', '5 MHz', ...}

File: trx_examples/streaming/LTE_PA_App/@LTEAppInternals/PlutoRadio.m [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+                case '20 MHz'
+                    obj.PlutoRx.SamplesPerFrame = 2^21;
+                otherwise
+                    st = dbstack;
+                    error("unknown option %s in %s", app.BWDropDown.Value, st.name);
+            end
+        end
+    else
+        switch (app.BWDropDown.Value)
+            case '3 MHz'
```
Comment (tfcollins) [inline trx_examples/streaming/LTE_PA_App/@LTEAppInternals/PlutoRadio.m:82]: Same as above

Comment (SrikanthPagadarai) [reply trx_examples/streaming/LTE_PA_App/@LTEAppInternals/PlutoRadio.m]: The latest commits address all of the comments.

File: trx_examples/streaming/LTE_PA_App/genLTEPAInstaller.m [hunk 6]
```diff
@@ -0,0 +1,28 @@
+cd(fileparts((mfilename('fullpath'))));
+p = pwd;
+
+version = '19.2.1';
```
Comment (tfcollins) [inline trx_examples/streaming/LTE_PA_App/genLTEPAInstaller.m:4]: Update to 20.1.1

File: trx_examples/streaming/LTE_PA_App/tests/testAppGUI.m [hunk 7]
```diff
@@ -0,0 +1,125 @@
+classdef testAppGUI < matlab.uitest.TestCase
```
Comment (tfcollins) [inline trx_examples/streaming/LTE_PA_App/tests/testAppGUI.m:1]: Class names start with capital letters

File: trx_examples/streaming/LTE_PA_App/@LTEAppInternals/LTEAppInternals.m (hunk 1 above)
Comment (tfcollins) [reply trx_examples/streaming/LTE_PA_App/@LTEAppInternals/LTEAppInternals.m:188]: Please crush these into a set loop like the others

**Comment by tfcollins** [comment]
Can you update the demo tests script to inherit from the test class added here.

**Pull request merged**